### ISSUE TITLE
Compilation issues

### ISSUE THREAD

**Issue by bjorn** [aurora-rs/stellar-base-rs#9]
I just tried adding `stellar-horizon = "0.6.0"` to my project, which depends on `stellar-base = "0.5.0"`. But compilation failed with the following messages:
```
error[E0599]: no function or associated item named `from_slice` found for struct `sodiumoxide::crypto::sign::Signature` in the current scope
  --> /home/bjorn/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stellar-base-0.5.0/src/signature.rs:60:39
   |
60 |         let sig = ed25519::Signature::from_slice(sb).ok_or(Error::InvalidSignature)?;
   |                                       ^^^^^^^^^^ function or associated item not found in `Signature`

error[E0616]: field `0` of struct `sodiumoxide::crypto::sign::Signature` is private
  --> /home/bjorn/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stellar-base-0.5.0/src/signature.rs:66:18
   |
66 |         self.sig.0.len()
   |                  ^ private field

error[E0616]: field `0` of struct `sodiumoxide::crypto::sign::Signature` is private
  --> /home/bjorn/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stellar-base-0.5.0/src/signature.rs:76:18
   |
76 |         self.sig.0.to_vec()
   |                  ^ private field

error[E0616]: field `0` of struct `sodiumoxide::crypto::sign::Signature` is private
  --> /home/bjorn/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stellar-base-0.5.0/src/signature.rs:81:19
   |
81 |         &self.sig.0
   |                   ^ private field

```
It appears a change in `sodiumoxide` might have broken compilation. My `Cargo.lock` file indicates version 0.2.7 was chosen.

I could work around the issue by pinning `sodiumoxide` at version `0.2.6` by adding `sodiumoxide = "=0.2.6"` to my `Cargo.toml`, confirming that this is an issue introduced by `sodiumoxide` version 0.2.7.

**Comment by bjorn** [comment]
In other news, it appears that `sodiumoxide` [has been deprecated](https://github.com/sodiumoxide/sodiumoxide/commit/5bb1dfd2578539b89ffb0cbea25f21f00cfb963e) 2 years ago, so it might be time to look for a replacement. :-)

**Comment by bjorn** [comment]
I see these compilation issues have already been fixed in 64cb3f177743651185feb629f8b03c7246bb1a84 by @salessandri. Is there any chance of passing on maintainership of this library, in case you no longer have time for it, @fracek?

**Comment by fracek** [comment]
Hey, yes I'm happy to add someone as admin to help keeping the library alive.

**Comment by bjorn** [comment]
Hmm, I had hoped @salessandri might be interested, but it seems they may also ended up busy with other projects.

My project (Raccoin - https://raccoin.org) depends on `stellar-base` and `stellar-horizon` for importing transactions from Stellar accounts. Currently, I'm pointing the dependencies to Santiago's fork. However, that does not work when I want to publish my app also as a crate, since then it needs its dependencies to be published there as well.

I don't really have time to maintain these crates myself, but I could at least publish the updates that are already done, if you don't have time for it, @fracek.

**Comment by fracek** [comment]
Yes, I'm happy to add you as admin both here (so you can push changes from the fork you're using now) and on crates so you can publish an update there. Let me know if you're happy with that and I'll do it.

**Comment by bjorn** [comment]
@fracek Thanks! I'll push those updates and let's hope for the future there will be more interested in helping with maintenance. :-)

**Comment by fracek** [comment]
I added you as admin to the repo and org!

**Comment by bjorn** [comment]
@fracek Thanks, I'll have a look at making some new releases based on the work of @salessandri soon!

**Comment by fracek** [comment]
I added you on crates.io so you can make a clean release.

**Comment by bjorn** [comment]
> I added you on crates.io so you can make a clean release.

Thanks!

I have in the meantime merged the current `master` into the PR #6, resolved resulting issues and fixed the coverate CI job. Even if you're short on time, it would be great if you could give it a quick sanity check before I merge (or you can merge if you're fine with it). I would not squash this time, due to the multiple authors and the involved merge.

And regarding this issue, I see that it was already resolved in 1f1a35372f156d63df962ea4cb0b746ae84ec34e by @trevor-crypto as well, that just never made it into a release. But it means we can close this one. :-)
